File: test-plugins/org.yakindu.sct.generator.csharp.test/test-gen/SameNameDifferentRegionTest.cs
/**
* Copyright (c) 2016 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
/**
 *  Unit TestCase for SameNameDifferentRegion
 */
using System;
namespace Yakindu.SCR.SameNameDifferentRegion {
	using NUnit.Framework;
	using Yakindu.SCR;
	
	[TestFixture]
	public class SameNameDifferentRegionTest {
		private SameNameDifferentRegionStatemachine sct;
	
	[SetUp]
	public void setup() {
		sct = new SameNameDifferentRegionStatemachine();
		sct.init();
	}
	
		[Test]
		public void testsameNameDifferenRegionTest(){
			sct.enter();
			Assert.IsTrue(sct.isStateActive(SameNameDifferentRegionStatemachine.State.main_region_StateA));
			sct.raiseE1();
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(SameNameDifferentRegionStatemachine.State.main_region_StateB));
			Assert.IsTrue(sct.isStateActive(SameNameDifferentRegionStatemachine.State.main_region_StateB_r1_StateA));
			sct.raiseE1();
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(SameNameDifferentRegionStatemachine.State.main_region_StateB));
			Assert.IsTrue(sct.isStateActive(SameNameDifferentRegionStatemachine.State.main_region_StateB_r1_StateB));
		}
	}
}

File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/SameNameDifferentRegionTest/SameNameDifferentRegionTest.cc
#include <string>
#include "gtest/gtest.h"
#include "SameNameDifferentRegion.h"
#include "sc_runner.h"
#include "sc_types.h"

namespace  {

SameNameDifferentRegion* statechart;



//! The timers are managed by a timer service. */
static SctUnitRunner * runner;

class SameNameDifferentRegionTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new SameNameDifferentRegion();
		statechart->init();
		runner = new SctUnitRunner(
			statechart,
			false,
			200
		);
	}
	virtual void TearDown() {
		delete statechart;
		delete runner;
	}
};


TEST_F(SameNameDifferentRegionTest, sameNameDifferenRegionTest) {
	
	statechart->enter();
	
	EXPECT_TRUE(statechart->isStateActive(SameNameDifferentRegion::main_region_StateA));
	
	statechart->getDefaultSCI()->raise_e1();
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(SameNameDifferentRegion::main_region_StateB));
	
	EXPECT_TRUE(statechart->isStateActive(SameNameDifferentRegion::main_region_StateB_r1_StateA));
	
	statechart->getDefaultSCI()->raise_e1();
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(SameNameDifferentRegion::main_region_StateB));
	
	EXPECT_TRUE(statechart->isStateActive(SameNameDifferentRegion::main_region_StateB_r1_StateB));
	
	
}

}

